frame/executive/src/lib.rs

```diff
@@ -711,9 +711,26 @@ mod tests {
 		});
 	}
 
+	#[test]
+	fn unsigned_weight_is_noted_when_applied() {
+		let xt = TestXt::new_unsigned(Call::Balances(BalancesCall::set_balance(33, 69, 69)));
+		let len = xt.clone().encode().len() as u32;
+		let mut t = new_test_ext(1);
+		t.execute_with(|| {
+			assert_eq!(<frame_system::Module<Runtime>>::all_extrinsics_weight(), 0);
+			assert_eq!(<frame_system::Module<Runtime>>::all_extrinsics_len(), 0);
+
+			// This is okay -- balances transfer will panic since it requires ensure_signed.
+			assert_eq!(Executive::apply_extrinsic(xt), Ok(Err(DispatchError::BadOrigin)));
+
+			assert_eq!(<frame_system::Module<Runtime>>::all_extrinsics_weight(), len);
```

Comment:
but how comes that `weight` is equal to `len`?
maybe note it for future generations?

Reply:
It is noted in the same file in another test 
https://github.com/paritytech/substrate/blob/d43b9f3ff94eb1f914ef77b00489bb42e5e40424/frame/support/src/weights.rs#L316-L327